numpy/lib/stride_tricks.py

```diff
@@ -35,7 +35,18 @@ def _maybe_view_as_subclass(original_array, new_array):
     return new_array
 
 
-def as_strided(x, shape=None, strides=None, subok=False, writeable=True):
+def _actual_head(x):
+    return sum((sh - 1) * (st if st < 0 else 0)
```

Comment:
Dunno, but this code smells a bit like it might go bad if you have an empty array. There is also `np.byte_bounds` which seems similar, just to note though, dunno if its worth using it or so.